.ci-builds/.azure-valgrind.yml

```diff
@@ -51,9 +51,115 @@ jobs:
         CC: clang
         CXX: clang++
     - script: |
-        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest
-      displayName: 'Valgrind intltest'
-      timeoutInMinutes: 60
+        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest bidi
+      displayName: 'Valgrind intltest bidi'
+      timeoutInMinutes: 10
+    - script: |
+        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest collate
+      displayName: 'Valgrind intltest collate'
+      timeoutInMinutes: 10
+    - script: |
+        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest convert
+      displayName: 'Valgrind intltest convert'
+      timeoutInMinutes: 10
+    - script: |
+        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest csdet
+      displayName: 'Valgrind intltest csdet'
+      timeoutInMinutes: 10
+    - script: |
+        cd icu4c/source/test/intltest && LD_LIBRARY_PATH=../../lib:../../stubdata:../../tools/ctestfw:$LD_LIBRARY_PATH valgrind --tool=memcheck --error-exitcode=1 --leak-check=full --show-reachable=yes ./intltest format
+      displayName: 'Valgrind intltest format'
+      timeoutInMinutes: 10
```

Comment:
It seems this is taking more than 10 minutes in the [pipeline](https://dev.azure.com/ms/icu/_build/results?buildId=433382&view=logs&jobId=1c430cc3-26fc-5b18-b02b-554388c8d5bc&j=1c430cc3-26fc-5b18-b02b-554388c8d5bc&t=ecab222d-d354-54a5-9ddd-1dec6d8bd2d7).
I think we might have to increase the time limit @FrankYFTang